rfcs/20210726-pooling-with-padding-value/README.md

```diff
@@ -0,0 +1,194 @@
+# RFC: Pooling with padding value
+
+## Introduction
+
+This document introduces new feature of Pooling operation that allows user to
+customize padding value used in the operation.
+
+### Motivation
+
+The original request comes from OpenVino team. OpenVino updates definition of
+MaxPool operation (more details: [MaxPool-8](https://github.com/openvinotoolkit/openvino/pull/5359)).
+Part of this request is to add parameter `pads_value`:
+* *pads_value*
+    * **Description**: *pads_value* will be used to fill pixels added during padding.
+      If not specified, the new pixels will be filled in with minimum value for the type.
+    * **Range of values**: floating point values
+    * **Type**: float
+    * **Default value**: *lowestinity*
+    * **Required**: *no*
+This request downstreams to oneDNN since OpenVino uses oneDNN-based plugin.
+
+An additional request is to add padding value support in Pooling as a
+performance optimization to cover `Pad + Pooling` fusion.
+In OpenVino specification Pad operation supports:
+•   not only a constant padding value, but other options where padding value depend on the actual data in tensor;
+•   padding over non-spatial dimensions
+
+### Definition
+
+Current definition:
+
+```
+dst(n, c, oh, ow) = max(lowest, limits_{kh, kw} (src(n, c, oh * SH + kh * (DH + 1) - PH_L, ow * SW + kw * (DW + 1) - PW_L)))
+```
+
+New definition:
+
+```
+dst(n, c, oh, ow) = max(pad_value, limits_{kh, kw} (src(n, c, oh * SH + kh * (DH + 1) - PH_L, ow * SW + kw * (DW + 1) - PW_L)))
```

Comment:
How does this work for training in case all values in the window are negative for `pad_value = 0`? What do we put in a workspace?

Reply:
it doesn't work for training. In this case padding value is not defined and OpenVino don't plan to use this optimization in that case. I updated the RFC so it contains information about this and some other issues. Thanks for the question!